server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/utils/StarTreeQueryHelper.java

```diff
@@ -0,0 +1,248 @@
+    }
+
+    /**
+     * Gets StarTreeQueryContext from the search context and source builder.
+     * Returns null if the query and aggregation cannot be supported.
+     */
+    public static StarTreeQueryContext getStarTreeQueryContext(SearchContext context, SearchSourceBuilder source) throws IOException {
+        // Current implementation assumes only single star-tree is supported
+        CompositeDataCubeFieldType compositeMappedFieldType = (CompositeDataCubeFieldType) context.mapperService()
+            .getCompositeFieldTypes()
+            .iterator()
+            .next();
+        CompositeIndexFieldInfo starTree = new CompositeIndexFieldInfo(
+            compositeMappedFieldType.name(),
+            compositeMappedFieldType.getCompositeIndexType()
+        );
+
+        for (AggregatorFactory aggregatorFactory : context.aggregations().factories().getFactories()) {
+            MetricStat metricStat = validateStarTreeMetricSupport(compositeMappedFieldType, aggregatorFactory);
+            if (metricStat == null) {
+                return null;
+            }
+        }
+
+        // need to cache star tree values only for multiple aggregations
+        boolean cacheStarTreeValues = context.aggregations().factories().getFactories().length > 1;
+        int cacheSize = cacheStarTreeValues ? context.indexShard().segments(false).size() : -1;
+
+        return StarTreeQueryHelper.tryCreateStarTreeQueryContext(starTree, compositeMappedFieldType, source.query(), cacheSize);
+    }
+
+    /**
+     * Uses query builder and composite index info to form star-tree query context
+     */
+    private static StarTreeQueryContext tryCreateStarTreeQueryContext(
+        CompositeIndexFieldInfo compositeIndexFieldInfo,
+        CompositeDataCubeFieldType compositeFieldType,
+        QueryBuilder queryBuilder,
+        int cacheStarTreeValuesSize
+    ) {
+        Map<String, Long> queryMap;
+        if (queryBuilder == null || queryBuilder instanceof MatchAllQueryBuilder) {
+            queryMap = null;
+        } else if (queryBuilder instanceof TermQueryBuilder) {
+            // TODO: Add support for keyword fields
+            if (compositeFieldType.getDimensions().stream().anyMatch(d -> d.getDocValuesType() != DocValuesType.SORTED_NUMERIC)) {
+                // return null for non-numeric fields
+                return null;
+            }
+
+            List<String> supportedDimensions = compositeFieldType.getDimensions()
+                .stream()
+                .map(Dimension::getField)
+                .collect(Collectors.toList());
+            queryMap = getStarTreePredicates(queryBuilder, supportedDimensions);
+            if (queryMap == null) {
+                return null;
+            }
+        } else {
+            return null;
+        }
+        return new StarTreeQueryContext(compositeIndexFieldInfo, queryMap, cacheStarTreeValuesSize);
```

Comment:
nit : `return new StarTreeQueryContext(compositeIndexFieldInfo, queryMap, cacheStarTreeValuesSize);` can be moved up ( below call to getStarTreePredicates ) to avoid having
```
f (queryMap == null) {
                return null;
            }
```